# 1. Bayes' theorem
Bayes' theorem says P(A|B)=P(B|A)\*P(A)/P(B)

a)
60% of the kids play football, and 36% of the kids play ice hockey. 40% of the kids who play football
also play ice hockey. What percent of those that play ice hockey also play football?

a = football
b = hockey

inserting into the bayes theorem we get:


In [4]:
footballHockey = 0.4
football = 0.6
hockey = 0.36
footballHockey*football/hockey

0.6666666666666666

so 66.67 % of the children that play hockey, also play football.

b)

40% of the kids like music, and 24% of the kids like to dance. Given that 30% of those that like
music also likes to dance, what percent of those that like to dance also likes music?

a = music
b = dance

inserting into the bayes theorem we get:

In [3]:
music = 0.4
dance = 0.24
danceMusic = 0.3

danceMusic*music/dance

0.5

so 50% of the children that likes to dance also likes music.

c)

In a factory, machine X produces 60% of the daily output and machine Y produces 40% of the daily
output.
2% of machine X’s output is defective, and 1.5% of machine Y’s output is defective.
One day, an item is inspected at random, and found to be defective. What is the probability that it
was produced by machine X?

Initial probability of product being machine x = 60% = 0.6
Initial probability of product being machine y = 40% = 0.4

P A = P A|B1   *   P B1    +      P A|B2 * P B2


(P A - P A|B2 * P B2)/PB1 = P A|B1





In [13]:
0.6*0.02

0.012

In [14]:
 0.4*0.015

0.006

We are unsure how to calculate the probability for this one. But our guess is two thirds equal to 66.67% probability. We are guessing by taking the difference between the results above.

# 2. 
Make a KNN classifier on the IRIS dataset using Python. Make sure to split the dataset into training and
testing sets. 

In [1]:
from sklearn import datasets
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import train_test_split
from matplotlib import pyplot as plt
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score
from sklearn.metrics import auc

In [2]:
iris = datasets.load_iris()
X = iris.data
y= iris.target
knn = KNeighborsClassifier(n_neighbors=3)


X_train, X_test, y_train, y_test = train_test_split(iris.data, iris.target, train_size = 0.8, random_state = 5)

knn.fit(X_train,y_train)
y_predicted = knn.predict(X_test)

print(knn.score(X_test,y_predicted))

from sklearn.metrics import classification_report, confusion_matrix
print(confusion_matrix(y_test, y_predicted))
print(classification_report(y_test, y_predicted))

print(accuracy_score(y_test, y_predicted))

1.0
[[ 8  0  0]
 [ 0  9  2]
 [ 0  0 11]]
              precision    recall  f1-score   support

           0       1.00      1.00      1.00         8
           1       1.00      0.82      0.90        11
           2       0.85      1.00      0.92        11

    accuracy                           0.93        30
   macro avg       0.95      0.94      0.94        30
weighted avg       0.94      0.93      0.93        30

0.9333333333333333


As we can see from the code above, we get a 0.93 accuracy score with a training size of 0.8 and 3 neighbors.

# 3 - 5
See KNIME workflow.

# 6.
